packages/core/src/components/menu/menuItem.tsx

```diff
@@ -176,7 +176,13 @@ export class MenuItem extends AbstractComponent<IMenuItemProps, IMenuItemState>
             }
 
             let { left = 0, right = 0 } = this.props.submenuViewportMargin;
-            right = document.documentElement.clientWidth - right;
+            if (typeof document !== "undefined"
+                && typeof document.documentElement !== "undefined"
+                && Number(document.documentElement.clientWidth)) {
```

Comment:
Agreed. I spotted this, but unfortunately we can't test for `NaN` because `NaN == NaN` is never true. [`Number.isNaN()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN) would be perfect, but only works if you're targeting es6. Otherwise, `NaN` [correctly coerces to `false`](https://dorey.github.io/JavaScript-Equality-Table/), and the only other problematic value (coerces to a value we don't want) is 0, but if the `clientWidth` is 0 then it doesn't matter what the alignment is. So I left as-is. Only other option would be check if it's defined.